### PR TITLE
Allow more flexibility in choosing CR name

### PR DESCRIPTION
**Pull request by akalenyu**
Currently we are not very friendly in allowing a different CR name;
In the operator yaml we have hardcoded values, for instance:
https://github.com/kubevirt/hostpath-provisioner-operator/blob/main/deploy/operator.yaml#L170
and on the other hand, we refer to the SA in the code as "${cr.Name}-admin":
https://github.com/kubevirt/hostpath-provisioner-operator/blob/main/pkg/controller/hostpathprovisioner/controller.go#L467

This PR hardcodes the values of the resources in the code (treat operator.yaml as static).

- [x] Verify on u/s cluster
- [ ] Unit/Functional tests
- [ ] Come up with how to handle upgrade scenario 2.6 -> 4.8 (different PR?)

**Release note**:

```release-note
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1923243
```

Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

### CODE REVIEW

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 1]
```diff
@@ -88,16 +88,20 @@ var _ = Describe("Controller reconcile loop", func() {
 				Namespace: "test-namespace",
 			},
 		}
+		dsNN := types.NamespacedName{
+			Name:      "hostpath-provisioner",
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:92]: Use the const you just create in common.go?

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 2]
```diff
@@ -120,15 +124,15 @@ var _ = Describe("Controller reconcile loop", func() {
 			},
 		}
 		saNN := types.NamespacedName{
-			Name:      "test-name-admin",
+			Name:      "hostpath-provisioner-admin",
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:127]: Use the other const you just created in common.go

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 3]
```diff
@@ -120,15 +124,15 @@ var _ = Describe("Controller reconcile loop", func() {
...
 			Namespace: "test-namespace",
 		}
 		cr, r, cl = createDeployedCr(cr)
 		// Now modify the service account to something not desired.
 		sa := &corev1.ServiceAccount{}
 		err := cl.Get(context.TODO(), saNN, sa)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(sa.ObjectMeta.Labels["k8s-app"]).To(Equal("test-name"))
+		Expect(sa.ObjectMeta.Labels["k8s-app"]).To(Equal("hostpath-provisioner"))
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:135]: Use the const you just create in common.go?

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 4]
```diff
@@ -144,7 +148,7 @@ var _ = Describe("Controller reconcile loop", func() {
 		sa = &corev1.ServiceAccount{}
 		err = cl.Get(context.TODO(), saNN, sa)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(sa.ObjectMeta.Labels["k8s-app"]).To(Equal("test-name"))
+		Expect(sa.ObjectMeta.Labels["k8s-app"]).To(Equal("hostpath-provisioner"))
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:151]: Use the const you just create in common.go?

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 5]
```diff
@@ -155,7 +159,7 @@ var _ = Describe("Controller reconcile loop", func() {
 			},
 		}
 		croleNN := types.NamespacedName{
-			Name: "test-name",
+			Name: "hostpath-provisioner",
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:162]: Use the const you just create in common.go?

File: pkg/controller/hostpathprovisioner/contoller_test.go [hunk 6]
```diff
@@ -191,7 +195,7 @@ var _ = Describe("Controller reconcile loop", func() {
 			},
 		}
 		crbNN := types.NamespacedName{
-			Name: "test-name",
+			Name: "hostpath-provisioner",
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/contoller_test.go:198]: Use the const you just create in common.go?

File: pkg/controller/hostpathprovisioner/controller.go [hunk 7]
```diff
@@ -88,9 +88,20 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
...
+
+			if err := mgr.GetClient().List(context.TODO(), hppList, &client.ListOptions{}); err != nil {
+				log.Error(err, "Error listing HPPs")
+				return nil
+			}
+			if len(hppList.Items) == 0 {
+				log.Info("No HPPs in cluster")
+				return nil
+			}
+
```
Comment (awels) [inline pkg/controller/hostpathprovisioner/controller.go:102]: You will want to check for > 1 hpps as well. Which is also an error state.

**Comment by awels** [comment]
There are many more locations where you use the string where you could use the const.

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 1 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 2 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 3 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 4 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 5 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

File: pkg/controller/hostpathprovisioner/controller.go (hunk 7 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/controller.go:105]: Added this check as well, thx

File: pkg/controller/hostpathprovisioner/contoller_test.go (hunk 6 above)
Comment (akalenyu) [reply pkg/controller/hostpathprovisioner/contoller_test.go]: Done

**Pull request merged**